airmencoders/Logging-iOS
# PBS-1393 Replace AppDelegate and SceneDelegate

[PBS-1393](https://jira.il2.dso.mil/browse/PBS-1393)

**Discussion** 🗣
This PR updates the app to use the SwiftUI App() Lifecycle that was introduced in iOS 14. It replaces the old AppDelegate/SceneDelegate Lifecycle. 

**Important Notes** ✍️
- All instances of SceneDelegate have been removed from the app, where they were used to access the dataController ObservedObject. 
- There is now a single static instance of DataController that is accessed throughout the app where needed (which is initialized on launch).
- All scene state logic (active, inactive, background, etc.) is now handled by an Environment Enum called scenePhase.
- Both the AppDelegate and SceneDelegate files have been removed.

**Testing** ✅
- Launch the app in Sim and on Device.
- Make sure there are no issues/crashes on launch.
- Edit data in missions, terminate the app, and ensure the data persists across sessions.
